File: api/src/main/java/com/hirrao/javaee/mapper/HeartRateMapper.java
package com.hirrao.javaee.mapper;

import com.hirrao.javaee.entity.HeartRate;
import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import org.apache.ibatis.annotations.Insert;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Select;

import java.util.List;

@Mapper
public interface HeartRateMapper extends BaseMapper<HeartRate> {

    @Select("SELECT * FROM heartRateLog WHERE uid=#{uid}")
    List<HeartRate> searchHeartRateById(Long uid);
    @Insert("INSERT INTO heartRateLog(hrlId, uid, heartRate,recordTime) VALUES(#{hrlId},#{uid}, #{heartRate}, #{recordTime})")
    void insertHeartRate(long hrlId, long uid, float heartRate,String recordTime);
}

File: api/src/main/kotlin/cn/hirrao/javaee/mapper/HeartRateMapper.kt
package cn.hirrao.javaee.mapper

import cn.hirrao.javaee.entity.HeartRate
import com.baomidou.mybatisplus.core.mapper.BaseMapper
import org.apache.ibatis.annotations.Insert
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Select

@Mapper
interface HeartRateMapper : BaseMapper<HeartRate?> {
    @Select("SELECT * FROM heartRateLog WHERE uid=#{uid}")
    fun searchHeartRateById(uid: Long?): List<HeartRate?>?

    @Insert("INSERT INTO heartRateLog(hrlId, uid, heartRate,recordTime) VALUES(#{hrlId},#{uid}, #{heartRate}, #{recordTime})")
    fun insertHeartRate(hrlId: Long, uid: Long, heartRate: Float, recordTime: String?)
}
